test/functional/01_actions/bookshelf/05_serializer_formats.spec.js

```diff
@@ -0,0 +1,901 @@
+import { afterAll, beforeAll, beforeEach, describe, expect, it, vi } from 'vitest'
+// import { omit } from 'lodash/fp'
+import Joint from '../../../../src'
+import projectAppModels from '../../../scenarios/project-app/model-config'
+import bookshelf from '../../../db/bookshelf/service'
+import { resetDB } from '../../../db/bookshelf/db-utils'
+
+let projectApp = null
+let projectAppJsonApi = null
+let projectAppFlat = null
+
+// -----------------------------------------------------------------------------
+// SERIALIZER FORMATS [bookshelf]
+// -----------------------------------------------------------------------------
+describe('SERIALIZER FORMATS [bookshelf]', () => {
+  beforeAll(() => {
+    vi.useFakeTimers()
+    vi.setSystemTime(new Date('2024-01-01T00:00:00.000Z'))
+
+    // -----------
+    // Project App
+    // -----------
+    projectApp = new Joint({ service: bookshelf })
+    projectApp.generate({ modelConfig: projectAppModels, log: false })
+
+    projectAppJsonApi = new Joint({ service: bookshelf, output: 'json-api' })
+    projectAppJsonApi.generate({ modelConfig: projectAppModels, log: false })
+
+    projectAppFlat = new Joint({ service: bookshelf, output: 'flat' })
+    projectAppFlat.generate({ modelConfig: projectAppModels, log: false })
+  })
+
+  beforeEach(async () => {
+    vi.setSystemTime(new Date('2024-01-01T00:00:00.000Z'))
+    await resetDB(['users', 'roles', 'profiles', 'projects'])
+  })
+
+  afterAll(() => { vi.useRealTimers() })
+
+  describe('payload format: "json-api"', () => {
+    it('should return an item in JSON API shape (no associations)', async () => {
+      const specUser = {
+        modelName: 'User',
+        fields: [
+          { name: 'username', type: 'String', required: true }
+        ]
+      }
+      const inputUser = {
+        fields: {
+          username: 'ricksanchez'
+        }
+      }
+
+      const payload = await projectAppJsonApi.getItem(specUser, inputUser)
+
+      // Should contain only "data"
+      expect(payload).toHaveProperty('data')
+      expect(payload).not.toHaveProperty('included')
+
+      // Top level "data" keys
+      const dataTopLevelKeys = ['attributes', 'id', 'type']
+      expect(Object.keys(payload.data).sort()).toEqual(dataTopLevelKeys)
+
+      // Full "data" response
+      expect(payload).toMatchInlineSnapshot({
+        data: {
+          attributes: {
+            created_at: expect.any(Date),
+            updated_at: expect.any(Date),
+            last_login_at: expect.any(String)
```

Comment:
With `vi.setSystemTime()` I think you can safely remove the wildcard dates matcher?

Reply:
Oh. Right on. I will try that.